### PR TITLE
Detect conflicts between FQN classes and declared classes.

### PR DESCRIPTION
**Pull request by cburschka**
Fixing https://www.drupal.org/project/coder/issues/3137489

### CODE REVIEW

**Comment by cburschka** [comment]
(I'm not sure why the CI seems to be stuck at reporting back the build status)

File: coder_sniffer/Drupal/Sniffs/Classes/FullyQualifiedNamespaceSniff.php [hunk 1]
```diff
@@ -127,6 +127,21 @@ public function process(File $phpcsFile, $stackPtr)
             $useStatement = $phpcsFile->findNext(T_USE, ($endPtr + 1));
         }//end while
 
+        $classStatement = $phpcsFile->findNext(T_CLASS, 0);
+        while ($classStatement !== false) {
+            $afterClassStatement = $phpcsFile->findNext(T_WHITESPACE, $classStatement, null, true);
+            // Check for 'class ClassName' declarations.
+            if ($tokens[$afterClassStatement]['code'] === T_STRING) {
+                $declaredName = $phpcsFile->getTokensAsString($afterClassStatement, 1);
```
Comment (arkener) [inline coder_sniffer/Drupal/Sniffs/Classes/FullyQualifiedNamespaceSniff.php]: The token content `$tokens[$afterClassStatement]['content']` can be used to determine the class name.

File: coder_sniffer/Drupal/Sniffs/Classes/FullyQualifiedNamespaceSniff.php [hunk 2]
```diff
@@ -127,6 +127,21 @@ public function process(File $phpcsFile, $stackPtr)
             $useStatement = $phpcsFile->findNext(T_USE, ($endPtr + 1));
         }//end while
 
+        $classStatement = $phpcsFile->findNext(T_CLASS, 0);
+        while ($classStatement !== false) {
```
Comment (arkener) [inline coder_sniffer/Drupal/Sniffs/Classes/FullyQualifiedNamespaceSniff.php]: We should probably skip this conflict check if we already received a $conflict earlier in this sniff.

**Review by arkener — CHANGES_REQUESTED** [review]
Looks good to me, just 2 minor things and then we can merge this. We should also add an extra test for this use-case on FullyQualifiedNamespaceUnitTest.

**Comment by arkener** [comment]
Not really sure why Travis isn't reporting the status here, but the [tests](https://travis-ci.org/github/pfrenssen/coder/builds/694241406) are completed successfully.

File: coder_sniffer/Drupal/Sniffs/Classes/FullyQualifiedNamespaceSniff.php (hunk 1 above)
Comment (cburschka) [reply coder_sniffer/Drupal/Sniffs/Classes/FullyQualifiedNamespaceSniff.php]: Ah, thanks for pointing that out.

**Comment by cburschka** [comment]
Hadn't noticed I was working on an old branch; merging 8.x-3.x.

**Review by arkener — APPROVED** [review]

**Comment by arkener** [comment]
Looks good, merged, thanks!

**Pull request merged**